docs/vsock.md

```diff
@@ -0,0 +1,115 @@
+# VSOCK support
+
+VSOCK provides a way for guest and host to communicate through a socket. VSOCK sockets support both stream and datagram types.
+
+The `virtio-vsock` is based on the [Firecracker](https://github.com/firecracker-microvm/firecracker/blob/main/docs/vsock.md) implementation, where additional details can be found.
+
+# Basic Terms
+
+## Predefined Values
+
+| CID | Description |
+|-----|-------------|
+| -1  | Random CID | 
+|  0  | Hypervisor | 
+|  1  | Loopback | 
+|  2  | Host | 
+
+## Socat specific Notes
+
+See the `DATA VALUES` section of `man socat` for the closer info on the data formats. For a quick reference:
+
+| Specifier | Datatype |
+|-----------|----------|
+|     B     | uint8_t  |
+|     S     | uint16_t |
+|     I     | uint32_t |
+
+
+Same man page tells also about the `SOCKET-LISTEN` specificator, where by the last `local-address` component is to contain `the data representation of a sockaddr structure without sa_family`. Consult `linux/vm_sockets.h` on the given system for the details of `struct sockaddr_vm` layout.
+
+Socat `>=1.7.4` supports `VSOCK-CONNECT` and `VSOCK-LISTEN` address types directly, thus simplifying the usage and eliminating the need to pass a manually crafted `sockaddr` struct data.
```

Comment:
Is it worth mentioning about the old way of using `socat` given version >=1.7.4 supports VSOCK directly?